drivers/display/ssd1306.c

```diff
@@ -373,23 +376,36 @@ static int ssd1306_set_orientation(const struct device *dev,
 static int ssd1306_set_pixel_format(const struct device *dev,
 				    const enum display_pixel_format pf)
 {
-	if (pf == PIXEL_FORMAT_MONO10) {
-		return 0;
+	struct ssd1306_data *data = dev->data;
+
+	if (0U == (pf & (PIXEL_FORMAT_MONO10 | PIXEL_FORMAT_MONO01))) {
+		LOG_ERR("Unsupported");
+		return -ENOTSUP;
 	}
-	LOG_ERR("Unsupported");
-	return -ENOTSUP;
+	if (pf != data->current_format) {
+		data->current_format = pf;
+		uint8_t cmd_buf[] = {((pf == PIXEL_FORMAT_MONO10) ? SSD1306_SET_REVERSE_DISPLAY
+								  : SSD1306_SET_NORMAL_DISPLAY)};
+		if (ssd1306_write_bus(dev, cmd_buf, sizeof(cmd_buf), true)) {
+			return -EIO;
+		}
+	}
+	return 0;
```

Comment:
```c
	uint8_t cmd;
	int ret;

	if (pf == data->pf) {
		return 0;
	}

	if (pf == PIXEL_FORMAT_MONO10) {
		cmd = SSD1306_SET_REVERSE_DISPLAY;
	} else if (pf == PIXEL_FORMAT_MONO01) 
		cmd = SSD1306_SET_REVERSE_DISPLAY;
	} else {
		return -ENOTSUP;
	}

	ret = ssd1306_write_bus(dev, &cmd, 1, true);
	if (ret) {
		return ret;
	}

	data->pf = pf;

	return 0;
```

Reply:
thanks. I've changed it to your suggestion.